paperscraper/lib.py

```diff
@@ -324,21 +324,28 @@ async def parse_google_scholar_metadata(
     """Parse raw paper metadata from Google Scholar into a more rich format."""
     doi: str | None = paper["externalIds"].get("DOI", None)
     if doi:
-        bibtex = await doi_to_bibtex(doi, session)
-        key: str = bibtex.split("{")[1].split(",")[0]
-        citation = format_bibtex(bibtex, key, clean=False)
-    else:
+        try:
+            bibtex = await doi_to_bibtex(doi, session)
+            key: str = bibtex.split("{")[1].split(",")[0]
+            citation = format_bibtex(bibtex, key, clean=False)
+        except DOINotFoundError:
+            doi = None
+    if not doi:
         # get citation by following link
         # SLOW SLOW Using SerpAPI for this
         async with session.get(
             paper["inline_links"]["serpapi_cite_link"]
             + f"&api_key={os.environ['SERPAPI_API_KEY']}"
         ) as r:
+            # we raise here, because something really is wrong.
             r.raise_for_status()
             data = await r.json()
         citation = next(c["snippet"] for c in data["citations"] if c["title"] == "MLA")
         bibtex_link = next(c["link"] for c in data["links"] if c["name"] == "BibTeX")
         async with session.get(bibtex_link) as r:
+            # we may have a 443 - link expired
+            if r.status == 443:  # noqa: PLR2004
+                raise RuntimeError(f"443 on bibtex link at {bibtex_link}")
```

Comment:
This change is interesting to me, why special-case 443?  Is some caller specifically checking for `RuntimeError` in this event?

From reading around, I see we `raise_for_status` on the line below, why not just let that blow us up like this:

```python
try:
    r.raise_for_status()
    bibtex = await r.text()
except HTTPError as exc:
    raise RuntimeError("Failed on bibtex link {bibtex_link}.") from exc
```

It will give us a richer stack trace, without special casing just 443

Reply:
This is meant to check for a specific error - google scholar blocking bibtex links that we need to monitor. Sometimes they all fail and I'm not sure if it's IP block. Added note